pkg/genyaml/testdata/pointer_types/pointer_types.yaml

```diff
@@ -0,0 +1,12 @@
+# Employees is an array of employees of the zoo
+employees:
+  - # Age of employee
+    Age: 22
+
+    # Name of employee
+    Name: Jim
+  - # Age of employee
```

Comment:
I think this newline behaviour is hard to read. I'd group the list elements together without newlines (and maybe in general avoid newlines between map elements?), and maybe put a newline between list entries (which of course devolves in nested lists...).

How hard is it to change this behaviour?

Reply:
This is the *default* behavior of the [`yaml`](https://godoc.org/gopkg.in/yaml.v3) encoder. I will delve into the source code and documentation to gauge the control over and difficulty of this level of customization.

Reply:
It would appear this is an artifact of how the encoder adds `HeadComment` to the YAML output. Short of forking `go-yaml` at `v3` and changing this logic directly, which may actually not be a bad idea, it would be *difficult* to customize. 

With that being said, to get the most control over the encoding and formatting, this functionality would ideally live *inside* the encoder. For that reason, I have forked `go-yaml` and plan to evaluate the effort in building both the commented YAML generation as well as formatting directly into the library. This would be more of a generalized tool; one that could be ubiquitously useful. For this specific feature, is this format acceptable? 

An immediate alternative can be to run a simple YAML beautifier/formatter on the output.

Reply:
This behaviour is fine for now, if not ideal.

Reply:
I made a note of this in the README.md. I think we can get exactly what we want with a modified fork of `go-yaml` which will be a follow-up in the near future to this package.